app/src/main/java/org/xbmc/kore/ui/generic/SubtitlesOptionsDialog.java

```diff
@@ -0,0 +1,313 @@
+
+    private static final String TAG = LogUtils.makeLogTag(SubtitlesOptionsDialog.class);
+
+    /**
+     * List of available subtitles
+     */
+    private List<PlayerType.Subtitle> availableSubtitles;
+    private int currentSubtitleIndex = -1;
+
+    // Number of explicitly added options for subtitles (to subtract from the
+    // number of subtitles returned by Kodi)
+    static final int ADDED_SUBTITLE_OPTIONS = 3;
+
+    /**
+     * Constants for the general select dialog
+     */
+    private final static int SELECT_SUBTITLES = -111;
+
+    /**
+     * Host manager from which to get info about the current XBMC
+     */
+    private HostManager hostManager;
+
+    /**
+     * Handler on which to post RPC callbacks
+     */
+    private Handler callbackHandler = new Handler();
+
+    /**
+     * Activity to communicate potential actions that change what's playing
+     */
+    private HostConnectionObserver hostConnectionObserver;
+
+    /**
+     * The current active player id
+     */
+    private int currentActivePlayerId = -1;
+
+    /**
+     * Default callback for methods that don't return anything
+     */
+    private ApiCallback<String> defaultStringActionCallback = ApiMethod.getDefaultActionCallback();
+
+
+
+    private FragmentActivity dialogActivity;
+
+    public SubtitlesOptionsDialog(Activity activity){
+        hostManager =HostManager.getInstance(activity);
+        hostConnectionObserver=hostManager.getHostConnectionObserver();
+        hostConnectionObserver.registerPlayerObserver(this, true);
+        dialogActivity =(FragmentActivity)activity;
+    }
+
+
+    /**
+     * Display subtitles dialog
+     *
+     */
+    public void Show(){
+        int selectedItem = -1;
+
+        String[] subtitles = new String[(availableSubtitles != null) ?
+                availableSubtitles.size() + ADDED_SUBTITLE_OPTIONS : ADDED_SUBTITLE_OPTIONS];
+
+        subtitles[0] = dialogActivity.getString(R.string.download_subtitle);
+        subtitles[1] = dialogActivity.getString(R.string.subtitle_sync);
+        subtitles[2] = dialogActivity.getString(R.string.none);
+
+        if (availableSubtitles != null) {
+            for (int i = 0; i < availableSubtitles.size(); i++) {
+                PlayerType.Subtitle current = availableSubtitles.get(i);
+                subtitles[i + ADDED_SUBTITLE_OPTIONS] = TextUtils.isEmpty(current.language) ?
+                        current.name : current.language + " | " + current.name;
+                if (current.index == currentSubtitleIndex) {
+                    selectedItem = i + ADDED_SUBTITLE_OPTIONS;
+                }
+            }
+        }
+
+        GenericSelectDialog dialog = GenericSelectDialog.newInstance(this,
+                SELECT_SUBTITLES, dialogActivity.getString(R.string.subtitles), subtitles, selectedItem);
+        dialog.show(dialogActivity.getSupportFragmentManager(), null);
+
+    }
+
+    public void addListener(SubtitlesOptionsDialogListener toAdd) {
```

Comment:
Not clear why you need a listener. You add the `RemoteFragment` and `NowPlayingPanel` as listeners but they don't do anything when `onRemoteRequired()` is called.